src/els_execute_command_provider.erl

```diff
@@ -0,0 +1,52 @@
+-module(els_execute_command_provider).
+
+-behaviour(els_provider).
+
+-export([ handle_request/2
+        , is_enabled/0
+        , options/0
+        ]).
+
+-include("erlang_ls.hrl").
+
+%%==============================================================================
+%% els_provider functions
+%%==============================================================================
+
+-spec is_enabled() -> boolean().
+is_enabled() -> true.
+
+-spec options() -> map().
+options() ->
+  #{ commands => ["foo"] }.
+
+-spec handle_request(any(), els_provider:state()) ->
+  {any(), els_provider:state()}.
+handle_request({workspace_executecommand, Params}, State) ->
+  #{ <<"command">> := Command } = Params,
+  Arguments = maps:get(<<"arguments">>, Params, []),
+  Result = execute_command(Command, Arguments),
+  {Result, State}.
+
+%%==============================================================================
+%% Internal Functions
+%%==============================================================================
+
+-spec execute_command(string(), [any()]) -> [map()].
+execute_command(<<"replace-lines">>
+               , [#{ <<"uri">>   := Uri
+                   , <<"lines">> := Lines
+                   , <<"from">>  := LineFrom
+                   , <<"to">>    := LineTo }] = Arguments) ->
+  lager:info("execute_command:replace-lines  [Arguments=~p]", [Arguments]),
+  Method = <<"workspace/applyEdit">>,
+  Params = #{ edit =>
+                  els_text_edit:edit_replace_text(Uri, Lines, LineFrom, LineTo)
+            },
+  lager:info("execute_command:replace-lines  [Params=~p]", [Params]),
```

Comment:
I'm not sure if these log messages are necessary considering we have lowered the threshold and the logs now include `debug` as well.